node/narwhal/src/primary.rs

```diff
@@ -84,9 +84,9 @@ pub struct Primary<N: Network> {
 
 impl<N: Network> Primary<N> {
     /// Initializes a new primary instance.
-    pub fn new(account: Account<N>, storage: Storage<N>, ledger: Ledger<N>, dev: Option<u16>) -> Result<Self> {
+    pub fn new(account: Account<N>, storage: Storage<N>, ledger: Ledger<N>, ip: SocketAddr) -> Result<Self> {
```

Comment:
Please preserve the `dev: Option<u16>` syntax to be consistent with the library interface. This will be especially problematic in integration upstream and when storage DB files are needed.